From: OpenWrt AX1800 Pro build project
Subject: [PATCH] fstools: use ext4 for new RE-SS-01 overlays on OpenWrt 25.12

The image includes ext4 tools and kernel support. Keep detection of
existing filesystems, but use ext4 when initializing a new eMMC overlay.
This patch matches fstools 16718b6e3c0fc7db7be6ae5848db0eae88ac8a8b.

diff --git a/package/system/fstools/patches/100-force-ext4-overlay.patch b/package/system/fstools/patches/100-force-ext4-overlay.patch
new file mode 100644
--- /dev/null
+++ b/package/system/fstools/patches/100-force-ext4-overlay.patch
@@ -0,0 +1,79 @@
+diff --git a/libfstools/common.c b/libfstools/common.c
+--- a/libfstools/common.c
++++ b/libfstools/common.c
+@@ -126,45 +126,6 @@
+ 	return FS_NONE;
+ }
+ 
+-static bool use_f2fs(struct volume *v, uint64_t offset, const char *bdev)
+-{
+-	char typeparam[BUFLEN];
+-	uint64_t size = 0;
+-	bool ret = false;
+-	int fd;
+-
+-	if (get_var_from_file("/proc/cmdline", "fstools_overlay_fstype", typeparam, sizeof(typeparam))) {
+-		if (!strcmp("f2fs", typeparam))
+-			return true;
+-		else if (!strcmp("ext4", typeparam))
+-			return false;
+-		else if (strcmp("auto", typeparam))
+-			ULOG_ERR("unsupported overlay filesystem type: %s\n", typeparam);
+-	}
+-
+-	fd = open(bdev, O_RDONLY);
+-	if (fd < 0)
+-		return false;
+-
+-	if (ioctl(fd, BLKGETSIZE64, &size) == 0)
+-		ret = size - offset > F2FS_MINSIZE;
+-
+-	close(fd);
+-
+-	return ret;
+-}
+-
+-static bool use_f2fs_compression(void)
+-{
+-	char typeparam[BUFLEN];
+-
+-	if (!get_var_from_file("/proc/cmdline", "fstools_overlay_compression_type",
+-			       typeparam, sizeof(typeparam)))
+-		return false;
+-
+-	return typeparam[0] && strcmp(typeparam, "none");
+-}
+-
+ int block_volume_format(struct volume *v, uint64_t offset, const char *bdev)
+ {
+ 	int ret = 0;
+@@ -173,6 +134,9 @@
+ 	int fd;
+ 	__u32 deadc0de;
+ 	size_t sz;
++
++	(void)offset;
++	(void)bdev;
+ 
+ 	switch (volume_identify(v)) {
+ 	case FS_DEADCODE:
+@@ -216,17 +180,8 @@
+ 		}
+ do_format:
+ 		ULOG_INFO("overlay filesystem in %s has not been formatted yet\n", v->blk);
+-		if (use_f2fs(v, offset, bdev)) {
+-			if (use_f2fs_compression()) {
+-				ULOG_INFO("f2fs overlay compression enabled\n");
+-				snprintf(str, sizeof(str),
+-					 "mkfs.f2fs -q -f -O extra_attr,compression -l rootfs_data %s", v->blk);
+-			} else {
+-				snprintf(str, sizeof(str), "mkfs.f2fs -q -f -l rootfs_data %s", v->blk);
+-			}
+-		} else {
+-			snprintf(str, sizeof(str), "mkfs.ext4 -q -F -L rootfs_data %s", v->blk);
+-		}
++		/* This image ships ext4 support for the eMMC overlay. */
++		snprintf(str, sizeof(str), "mkfs.ext4 -q -F -L rootfs_data %s", v->blk);
+ 
+ 		ret = system(str);
+ 		break;
